### ISSUE TITLE
Error 1057  "dkimsigner went async did not call resume" after upgrade to 3.4

### ISSUE THREAD

**Issue by mcnabney** [Pro/dkim-exchange#368]
### Versions

- **Windows Server Version**: 2016
- **Exchange Version**: 15.1.2375.7
- **Installed DKIM Exchange Version**: 3.4.0

### Description

Dkimsigner went async did not call resume error... 325 errors in 4 days, probably each message sent. No trigger event determined. Began at 3:09 AM june 2nd. Upgrade from 3.3.x to 3.4 on May 27. No errors for 6 days, now constant errors

**Expected behavior:** [no errors]

**Actual behavior:** [errors]

**Comment by theklap** [comment]
Does this look like issue #362 ? https://github.com/Pro/dkim-exchange/issues/362

The fix was not put into a release and the ticket was closed and not set back to an open state.

**Comment by mcnabney** [comment]
> Does this look like issue #362 ?
> 
> The fix was not put into a release and the ticket was closed and not set back to an open state.

Yes this is #362. Performance is not suffering but error persists hundreds of times per week. Indications are to stay with 3.4 until further issue with bouncycastle is resolved?

**Comment by theklap** [comment]
> > Does this look like issue #362 ?
> > The fix was not put into a release and the ticket was closed and not set back to an open state.
> 
> Yes this is #362. Performance is not suffering but error persists hundreds of times per week. Indications are to stay with 3.4 until further issue with bouncycastle is resolved?

That is correct.

**Comment by DependableUser** [comment]
Is this still unresolved?
Maybe the bouncy castle dll issue has been fixed and it now downloads as it should?

**Comment by mcnabney** [comment]
I have not seen any indications that this issue has been resolved. I have just been waiting out the release of the next version in the hopes that it has been managed. I am still seeing the constant errors in the Exchange logs but I ignore them for now.

**Comment by brandonjank** [comment]
Several unresolved github issues led me here to this one. We are seeing the same errors on our Exchange 2019 CU12 servers.

Agent 'Exchange DkimSigner' went async but did not call Resume on the new thread, while handling event 'OnCategorizedMessage'
Event ID:	1057
Type:	Error	Event Log:	Application
Number:	551441	Computer:	EX2019B
Source:	MSExchange Extensibility	User:	n/a
Category:	MExRuntime

#215
#362
#368

**Comment by Skreabengt** [comment]
Implemented DkimSigner just recently was on smarthost before. Use Exchange 2019 CU12 Jan23SU, Build Number: 15.02.1118.021 and DkimSigner 3.4.0.  Get Event ID 1057 "Agent 'Exchange DkimSigner' went async but did not call Resume on the new thread, while handling event 'OnCategorizedMessage'.

Dkim signing is not always working as reported to mailhardener from Google:
 
![image](https://user-images.githubusercontent.com/122919016/212980259-e310b8d8-5557-4fdf-bd96-1761f7067bd3.png)

**Comment by jdthird** [comment]
Just saw the readme from May 2023, he's abandoned it since he moved to the hosted exchange, and has no more use for it.  His message included a note for who to reach out to if anyone wanted to take it over.  Guess it's dead.
